types/wlr_input_method_v2.c

```diff
@@ -20,10 +24,24 @@ static struct wlr_input_method_v2 *input_method_from_resource(
 	return wl_resource_get_user_data(resource);
 }
 
+static struct wlr_input_method_keyboard_grab_v2 *im_keyboard_grab_from_resource(
+		struct wl_resource *resource) {
+	assert(wl_resource_instance_of(resource,
+		&zwp_input_method_keyboard_grab_v2_interface,
+		&im_keyboard_grab_impl));
+	return wl_resource_get_user_data(resource);
+}
+
+static void im_keyboard_grab_destroy(
```

Comment:
This doesn't seem used.

Reply:
It becomes used now after I resolved previous conversation about inert design and resource destory.